crates/accelerate/src/synthesis/evolution/mod.rs

```diff
@@ -0,0 +1,431 @@
+// This code is part of Qiskit.
+//
+// (C) Copyright IBM 2024
+//
+// This code is licensed under the Apache License, Version 2.0. You may
+// obtain a copy of this license in the LICENSE.txt file in the root directory
+// of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+//
+// Any modifications or derivative works of this code must retain this
+// copyright notice, and modified files need to carry a notice indicating
+// that they have been altered from the originals.
+
+#![allow(clippy::too_many_arguments)]
+
+use crate::synthesis::clifford::greedy_synthesis::resynthesize_clifford_circuit;
+
+use pyo3::prelude::*;
+use pyo3::types::{PyList, PyString, PyTuple};
+use smallvec::{smallvec, SmallVec};
+use std::borrow::Borrow;
+
+use qiskit_circuit::circuit_data::CircuitData;
+use qiskit_circuit::operations::{multiply_param, radd_param, Param, StandardGate};
+use qiskit_circuit::Qubit;
+
+use rustiq_core::structures::{
+    CliffordCircuit, CliffordGate, IsometryTableau, Metric, PauliLike, PauliSet,
```

Comment:
Is there a check needed that the user installed rustiq before using this code and raise a warning otherwise?